packages/blade/src/components/TextArea/_decisions/decisions.md

```diff
@@ -0,0 +1,70 @@
+| isDisabled           | `boolean`                        | `false` | Marks the input field as disabled. It'll also set aria-disabled to true and hence the input field will loose keyboard focus                                                                                                                                  |          |
+| isRequired           | `boolean`                        | `false` | Marks the input field as required. if the errorText is provided then that will be shown if the field is required                                                                                                                                             |          |
+| neccessityIndicator  | `required, optional`             |         | Indicator to be shown next to the label for emphasizing the necessity of the field. if optional is provided then it'll show `(optional)` and if required is provided then it'll show `*`                                                                     |          |
+| showClearButton      | `boolean`                        | `true`  | Defines if we want to show the clear icon button on the right side of the password field                                                                                                                                                                     |          |
+| onClearButtonClicked | `() => {}`                       |         | Used in conjunction with showClearButton prop. If clicked it removes the contents of the TextArea. Users can also hook their own analytics code or do something else.                                                                                        |          |
+| maxLines             | `number`                         | `2`     | Sets the number of lines (or rows) of the text area, minimum is `2` maximum is `5`                                                                                                                                                                           |          |
+| maxCharacters        | `number`                         |         | Used to show a character counter under the input field                                                                                                                                                                                                       |          |
+| validationState      | `error , success, none`          |         | Applies the style to the input field based on validation status and shows either errorText or successText respectively                                                                                                                                       |          |
```

Comment:
can we have the warning state also

Reply:
https://github.com/razorpay/blade/pull/681#discussion_r971835438